public/js/gulliver.es6.js

```diff
@@ -278,13 +278,38 @@ function setupConfig() {
   }
 }
 
+function setupMessaging() {
+  const messaging = new Messaging();
+  messaging.init();
+
+  const checkbox = document.getElementById('notifications');
```

Comment:
A good way to propagate the notification state to the element handler could be registering the element handler as a listener to messaging.js. Messaging.js would fire events for status changes such as: permission denied, permission allowed, ...